torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -137,9 +135,9 @@ Reducer::Reducer(
   // we do not expect sparse gradients for any parameter.
   if (expect_sparse_gradients_.empty()) {
     expect_sparse_gradients_ = std::vector<std::vector<bool>>(
-        replicas_.size(), std::vector<bool>(replicas_[0].size(), false));
+        1, std::vector<bool>(replicas_.size(), false));
   }
-  TORCH_INTERNAL_ASSERT(expect_sparse_gradients_.size() == replicas_.size());
+  TORCH_INTERNAL_ASSERT(expect_sparse_gradients_.size() == 1);
```

Comment:
This should imply that `expect_sparse_gradients` should now be able to just be a single boolean right?

Reply:
Yep, I wasn't sure if the related nested containers (expect_sparse_gradients, grad_accumulators) should also be changed